proposal/AilurusUmbra/README.md

```diff
@@ -0,0 +1,136 @@
+![](https://miro.medium.com/max/1200/1*e_oU-f6hX4FSSD-1OukP6Q.png) Figure 3. [16]
+
+* Convolution filter size [17]
+  
+  More and more papers of Neural Architecture Search show that 5x5 convolution filter might give us better result than 3x3 filters [17]. This project would compare results of both 5x5 and 3x3 convolution filters.
+
+### Numerical Method to be Applied
+
+* Initialization and Normalization
+  * BatchNorm with random initialization
+  * SkipInit
+  
+* Optimization 
+  * SGD
+  * Adagrad
+  * Adam
+
+* Convolution Filter
+  * 5x5 window 
+  * 3x3 window
+  * 5x5 depthwise-separable
+  * 3x3 depthwise separable
+ 
+
+## Perspective users
+
+* BCI researchers for processing EEG/MEG.
+* Engineers that need to do multichannel time series prediction/classification.
+
+## System architecture
+
+Most of the computation would be leverage to Pytorch framework, so that tensors could be easily move to VRAM of GPU to compute, and move back to RAM after computation.
+Models would takes `torch.tensor` as data strcuture to compute, this could be preprocessed after load data from mne. `torch.tensor` is easy to transformed into `numpy.array` which is commonly used in mne.
+This project might not implement tensorflow version of codes, because there is a similar developing frameworks called [mneflow]() [4]
+
+
+API description
+===============
+
+* Models: Refer to [torchvision.models](https://pytorch.org/docs/stable/torchvision/models.html)
+
+  * eegnn.models.eegnet(pretrained=True, denoise=True, device='cuda', sample_rate=4, optimizer=sgd())
+  * eegnn.models.sccnet(pretrained=True, device='cpu', nthreads=1)
+  * eegnn.models.moblie_eegnet(pretrained=False, device='cuda', sample_rate=4,  optimizer=adam())
+
+  * SOTA Normalization techniques which could replace BatchNorm [11]
+    * models.skipinit()
+
+* Utils: Some supplementary functions could just use api of [mne](https://mne.tools/stable/index.html) to prevent from reinventing the wheel. For example, read FIF data (MEG) or load sample dataset.
+
+  * mne.io.read_raw_fif(fname[, allow_maxshield, …])            Reader function for Raw FIF data.
+  * mne.datasets.eegbci.load_data(subject, runs[, path, …])     Get paths to local EEGBCI dataset files.
+
+## Engineering infrastructure
+
+This project would be mainly developed by python>=3.6 and based on Pytorch>=1.0 and tested on Debian, Mac OS and Win10 for compatibility. If training models from scratch is needed, it is recommended that running EEGNN on nvidia GPUs with at least 6GB VRAM (e.g. GTX 1060) for less training time. For only fine-tuning pretrained models or just doing inference, EEGNN provides multi-thread mechanism for CPU computation speedup. EEGNN is designed for both high performance computing on training models or efficient inference on BCI portable devices.
```

Comment:
Validating and testing should be included in your engineering infrastructure.

Reply:
This is modified in this commit [d96a523](https://github.com/yungyuc/nsdhw_20sp/pull/67/commits/d96a523368017530f4807ab4acb4247de3fe1cd8)